ext/xmlwriter/php_xmlwriter.stub.php

```diff
@@ -90,13 +90,13 @@ class XMLWriter
 {
     /**
      * @return bool
-     * @alias xmlwriter_open_uri
+     * @static-method-alias xmlwriter_open_uri
      */
     public function openUri(string $uri) {}
```

Comment:
I noticed that this little bastard - and the next one as well - has a different return type based on whether we are in an object-context or not (it returns a new object if called as a function). What happens with the `getObject()` call if this instance method is called statically? When I marked it as `static`, many tests broke because of the uninitialized object.

Another question that came to my mind if we want to also check the return type of aliases? I think we should, but these special cases like this one could veto this, unless we exclude them from the check.

Reply:
I think it may make more sense to add a `@no-verify` tag or similar rather than relaxing the rules or adding more alias types. Clearly we have some methods where PHP is being PHP and odd things are done for historical reasons. We should just say that we're aware of them and ignore.

Reply:
OK, I added `@no-verify`, it will probably come in handy for 3rd parties. I made a few methods static (e.g. `mysqli::debug`) previously, so we should decide if we really want to make this change.

Besides, I implemented two other improvements:
- The return type of function and final method aliases are now validated. Unfortunately, I had to add a class-specific field to the `FuncInfo` class... which is really not that nice, I know, although I think it's ok to do in this case. I can try to eliminate it if you have doubts.
- The arginfo files will only be updated when the `--verify` option is passed, if there is no error. I think this behaviour is what makes sense. :)

Reply:
> OK, I added `@no-verify`, it will probably come in handy for 3rd parties. As I made a few methods static (e.g. `mysqli::debug`) previously, we should decide now if we rather mark them `@no-verify`. I have no strong preference.

I'm okay with the remaining `static` changes, they make sense. There is BC potential here for inheriting classes, but we have precedent for this kind of change in XMLReader.

> Besides, I implemented two other improvements:
> 
>     * The return type of function and final method aliases are now validated. Unfortunately, I had to add a class-specific field to the `FuncInfo` class... which is really not that nice, I know, although I think it's ok to do in this case. I can try to eliminate it if you have doubts.

Hm, I forgot about the issue with the non-final methods. In this case, what we probably really want to validate is the return type against the docblock return type. But those are not so directly comparable. I think it may be better leave off the return type validation for now.

>     * The arginfo files will only be updated if there is no error when the `--verify` option is passed. I think this behaviour is what makes sense. :)

TBH the behavior you had previously made more sense to me. You may want to test some changes that only fix part of the verification warnings.

Reply:
> Hm, I forgot about the issue with the non-final methods. In this case, what we probably really want to validate is the return type against the docblock return type. But those are not so directly comparable. I think it may be better leave off the return type validation for now.

Yes, I didn't want to complicate my life with parsing and storing the phpdoc, so that's the reason I skipped it. However, I don't exactly get your last sentence. Do you mean to leave off this check for all cases or just non-final methods?

> TBH the behavior you had previously made more sense to me. You may want to test some changes that only fix part of the verification warnings.

OK, I'll revert this.

Reply:
> > Hm, I forgot about the issue with the non-final methods. In this case, what we probably really want to validate is the return type against the docblock return type. But those are not so directly comparable. I think it may be better leave off the return type validation for now.
> 
> Yes, I didn't want to complicate my life with parsing and storing the phpdoc, so that's the reason I skipped it. However, I don't exactly get your last sentence. Do you mean to leave off this check for all cases or just non-final methods?

I would have left it off entirely, but as you already implemented it, we can as well keep it.

Reply:
Done with the revert. I'll add an upgrading note about the static method changes